sec2_the_ruby_programming_language/2.3_number.rb

```diff
@@ -0,0 +1,75 @@
+# Numbers
+puts 1 + 2       #3
+puts 2 - 1       #1
+puts 10 * 2      #20
+puts 10 / 4      #2
+puts 10.0 / 4    #2.5
+puts 10 / 4.0    #2.5
+puts 10 / 4.to_f #2.5
+puts (10/4).to_f #2.0
+
+# Assign variables
+x = 5
+y = 10
+puts y / x #2
+puts y % x #0
+
+# Numbers vs String Concatenation
+# "5" * "5"  //no implicit conversion of String into Integer
+puts "5" * 2  #55
+#2 * "5"     //String can't be coerced into Integer
+#Note: A string can be multiplied by a number but a number can't multiplied by a string
+puts "-" * 20 #--------------------
+
+#Using method
+20.times{ print "-"}       #--------------------
+20.times{ puts rand(10)}   #generates 20 random numbers between 0 and 9
+puts rand                       #generates random number between 0 and 1
+puts rand(100)                   #generates number in between 0 and 99
+
+#String to numbers
+x = "5".to_i
+puts x        #5
+puts x.to_f   #5.0
+
+#Convert a string that can't be converted into a number
+puts "hello".to_i #0
+puts "hello".to_f #0.00
+
+#Simple calculator
+puts "Simple calculator"
+25.times { print "-"}
+puts "\nEnter the first number"
+num_1 = gets.chomp
+puts "Enter the second number"
+num_2 = gets.chomp
+puts "The first number multiplied by the second number is #{num_1.to_i * num_2.to_i}" 
```

Comment:
Why `.to_i`? If you key in:
```
num_1 = 2.5
num_2 = 3
```
it will be wrong answer

Reply:
Understood. Will remove it from the code.

Reply:
This is because the `num_1` and `num_2` variables are assigned to input with string type. Hence, we will need to convert the `num_1` and `num_2` implicitly into integer types so that the multiplication can be carried out successfully.

Reply:
ok. So, you are telling me need to convert to to_i. I never refute that. I am saying by doing so, you will yield a wrong answer. So, no idea what you are talking about here

Reply:
Okay sure understood. Thank you.

Reply:
??

Reply:
Is the wrong answer here referring to 2.5 * 2 supposed to be 5 but it will result in 4 instead?

Reply:
huh?
no.
See my inputs above pls

Reply:
I see the problem when 2.5 * 3, the output given here is 6 instead of 7.5. I'll remove this code section.